eth_tester/main.py

```diff
@@ -244,15 +273,25 @@ def get_nonce(self, account, block_number="latest"):
     #
     # Blocks, Transactions, Receipts
     #
+
+    def _get_pending_transaction_by_hash(self, transaction_hash):
+        for transaction in self._pending_transactions:
+            if transaction['hash'] == transaction_hash:
+                return transaction
+
     def get_transaction_by_hash(self, transaction_hash):
         self.validator.validate_inbound_transaction_hash(transaction_hash)
-        raw_transaction_hash = self.normalizer.normalize_inbound_transaction_hash(
-            transaction_hash,
-        )
-        raw_transaction = self.backend.get_transaction_by_hash(raw_transaction_hash)
-        self.validator.validate_outbound_transaction(raw_transaction)
-        transaction = self.normalizer.normalize_outbound_transaction(raw_transaction)
-        return transaction
+        pending_transaction = self._get_pending_transaction_by_hash(transaction_hash)
```

Comment:
Rather than having this function return `None` when the transaction isn't found, I think it'd be better if it raised `TransactionNotFound`.  This way nobody *accidentally* returns `None` because they forgot to check the return value.

Reply:
Good point, it also makes the flow in `get_transaction_by_hash` easier to read